### PR TITLE
Fix parsing for remanga API

### PR DESCRIPTION
**Pull request by ilyasemenow90**
Looks like remanga has been updated the API response and the app fails to parse it (home page has an error message).
The response fails by parsing JSON because msg is missing.

### CODE REVIEW

File: ReManga/Api/ReManga/ReMangaApi.swift [hunk 1]
```diff
@@ -98,19 +98,22 @@ class ReMangaApi: ApiProtocol {
         }
 
         let url = "https://api.remanga.org/api/search/catalog/?count=30&ordering=-\(sorting.id)&page=\(page)\(tags)"
-        let model: ReMangaApiMangaCatalogResult = try await performRequest(makeRequest(url))
-
-        var bookmarks: [ApiMangaBookmarkModel] = []
-        do { bookmarks = try await fetchBookmarkTypes() }
-        catch { print(error) }
+        do {
+            let model: ReMangaApiMangaCatalogResult = try await performRequest(makeRequest(url))
```
Comment (ilyasemenow90) [inline ReManga/Api/ReManga/ReMangaApi.swift]: This line was failed. I've added in under `do catch` block to handle this in a future

File: ReManga/Api/ReManga/Models/ReMangaApiMangaCatalogResult.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 // MARK: - ApiMangaCatalog
 struct ReMangaApiMangaCatalogResult: Codable {
-    let msg: String
+    let msg: String?
```
Comment (ilyasemenow90) [inline ReManga/Api/ReManga/Models/ReMangaApiMangaCatalogResult.swift:12]: msg is missing in new remanga response

File: ReManga/Api/ReManga/ReMangaApi.swift (hunk 1 above)
Comment (XITRIX) [reply ReManga/Api/ReManga/ReMangaApi.swift]: Could you, please, remove `do catch` from here? Networking layer is not the place for such logic, it should throw Error as it is, and it's responsibility of ViewModel to handle it properly

**Comment by XITRIX** [comment]
Only one comment above, the rest looks good

File: ReManga/Api/ReManga/ReMangaApi.swift (hunk 1 above)
Comment (ilyasemenow90) [reply ReManga/Api/ReManga/ReMangaApi.swift]: @XITRIX Done!

**Comment by XITRIX** [comment]
Thanks a lot!

**Pull request merged**